packages/shared/lib/core/wallet/actions/buildWalletState.ts

```diff
@@ -29,31 +31,44 @@ export async function buildWalletState(
         delegations: [],
     }
 
-    let depositAddress = ''
     let votingPower = ''
     let walletOutputs: OutputData[] = []
     let accountOutputs: OutputData[] = []
     let implicitAccountOutputs: OutputData[] = []
 
     try {
         balances = await wallet.getBalance()
-        depositAddress = await getDepositAddress(wallet)
         votingPower = balances.baseCoin.votingPower
         accountOutputs = await wallet.accounts()
         implicitAccountOutputs = await wallet.implicitAccounts()
         walletOutputs = await wallet.outputs()
+
+        // initialize selectedWalletMainAccountId if there is none set so the wallet can be used
+        // TODO: check that selectedWalletMainAccountId is still an owned account
+        const _selectedWalletMainAccountId = get(selectedWalletMainAccountId)
+        const blockIssuerAccounts = await getBlockIssuerAccounts(wallet)
+
+        // check if the current selectedWalletMainAccountId is still owned by the wallet
+        if (
```

Comment:
I don't know if this is the place to handle this. I think this should be updated when an account output is spent, not when the wallet is being built